### PR TITLE
React Redux Group Project: Space-X

### PR DESCRIPTION
**Pull request by emhamza**
### I am so happy that my partner and I have managed to achieve this milestone. In this project, we have done the mentioned tasks:

- [x] Kanban setup
- [x] Creating a React app
- [x] Config Linter
- [x] Adding header or Navbar to our app
- [x] Fetching data for Rockets and Missions from an external API
- [x] Created My Profile as per requirements 
- [x] Styling of the Space-X app
- [x] Unit tests for each component and function

### CODE REVIEW

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @emhamza and @MarwanDev, 
# Status: Changes Required ♻️

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<div align='center'>

</div>

## Good Highlights✔️

- Good documentation in readme
- Proper state management with redux
- Good team collaboration
- Directories are well structures


## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Missions/Missions.jsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+} from "../../redux/missionSlice/missionsSlice";
+
+const Missions = () => {
+  const missions = useSelector((state) => state.mission.missions);
+  const { Loading, error } = useSelector((state) => state.mission);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getMissions());
+  }, [dispatch]);
```
[the remark this answers is not in the document]
Comment (emhamza) [reply src/components/Missions/Missions.jsx:26]: Certainly, we will do that, your feedback is very progressive for us 😃 👍

File: src/components/Rockets/Rocket.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      <div className="rocket-data">
+        <h2 className="roc-name">{rocketData.rocket_name}</h2>
+        {rocketData.reserved && (
+         <>
+            <p className="roc-text">
+              <Button className="cancel-btn" type="submit">Reserved</Button>
+              {rocketData.description}
+            </p>
+            <Button color="primary" onClick={cancelReservationHandler}>Cancel Reservation</Button>
+          </>
```
[the remark this answers is not in the document]
Comment (emhamza) [reply src/components/Rockets/Rocket.jsx:30]: Sure thing, we thought the styling would not be an issue in this project. But I am glad that you pointed that out @emmiiorji

**Review by youmari — APPROVED** [review]
Hi @emhamza @MarwanDev , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights 
✔️ Clean code 
✔️ PRO PR 
Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**